### DISCUSSION TITLE
How to match the prediction and targets bounding boxes for MAP

### DISCUSSION THREAD

**Discussion by FrancescoSaverioZuppichini** [Lightning-AI/torchmetrics#589] [CompVision] [answered]
Hi Guys,

I hope you are all doing great. I was wondering how to correctly use the new `MAP` metric. Since targets and predictions have to have the same len, I was wondering how to correctly preprocess the predicted bonding box before passing them to MAP. Since at any step the predicted bounding boxes may be more than the targets, how do I correctly filtered them out? Should I find the matching using iou?

Thanks

**Comment by tkupek** [comment, marked as answer]
Hi @FrancescoSaverioZuppichini and thanks for the question.

What might be misleading here is the fact that `targets` and `preds` is a list of dicts. Each dict represents a sample and contains a list of boxes. While the **number of samples** in `targets` and `preds` has to match, the lists of boxes can vary.

For an example I recommend to have look at this test:
https://github.com/PyTorchLightning/metrics/blob/master/tests/detection/test_map.py

Does this make sense to you?

**Comment by FrancescoSaverioZuppichini** [comment]
Oh Thanks! Got it ;)
